data-prepper-plugins/prometheus-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/prometheus/service/PrometheusTimeSeries.java

```diff
@@ -123,231 +150,203 @@ private void addLabel(String name, final Object value) {
     }
 
     private void addLabelSanitized(final String name, final Object value) {
-        String valueStr;
-        if (value instanceof String) {
-            valueStr = (String)value;
-        } else {
-            valueStr = String.valueOf(value);
-        }
+        String valueStr = (value instanceof String) ? (String) value : String.valueOf(value);
         Label label = Label.newBuilder().setName(name).setValue(valueStr).build();
         labels.add(label);
-        size += label.toByteArray().length;
+        size += estimateLabelSize(name, valueStr);
     }
 
-    public List<TimeSeries> getTimeSeriesList() {
-        return timeSeriesList;
+    private int estimateLabelSize(String name, String value) {
+        return name.length() + value.length() + 8; // Approximate protobuf overhead
```

Comment:
Are you looking for length or size? This may not give you the actual value you are looking for.

I appreciate the desire to cut the extra buffer. But, the calculation may be incorrect. Could you add to the `Remote.WriteRequest.Builder` and get the value at that point to avoid double buffering?